### PR TITLE
Ath-1792: Create user page 

### PR DESCRIPTION
**Pull request by EoinMcD**
# Description

- Creates a page at /user that has placeholder text describing future features
- Has tests to ensure the page is loading and that text is present.

# Testing

# Documentation Changes
# Dependant PRs

### CODE REVIEW

**Comment by alvarocavalcanti** [comment]
My first comment isn't a blocker in any way, but it's targeted more at best practices: this PR has 132 files. This is HUGE and it can be tricky to review. I'm about to start my review but my initial feedback is, to make sure that unnecessary files aren't added and re-assess the work division. Maybe there's  a lot of boilerplate code that could have been extracted into a separate task/PR?

File: README.md [hunk 1]
```diff
@@ -1 +1,24 @@
-# BloggingEngine
+# README
```
Comment (alvarocavalcanti) [inline README.md:1]: This is the initial state of this file, and ideally it should be updated by each PR adding the new instructions (if any).

For now, which of the sections below can already be provided?

File: app/assets/config/manifest.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+//= link_tree ../images
```
Comment (alvarocavalcanti) [inline app/assets/config/manifest.js:1]: Is this file really needed? It looks like dead code.

File: app/assets/stylesheets/application.css [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/*
```
Comment (alvarocavalcanti) [inline app/assets/stylesheets/application.css:1]: This looks like a boilerplate file, which could be deleted and only added when needed. But I don't think that's mandatory, we can leave it be for now as a future guidance.

File: app/javascript/packs/hello_react.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// Run this example by adding <%= javascript_pack_tag 'hello_react' %> to the head of your layout file,
```
Comment (alvarocavalcanti) [inline app/javascript/packs/hello_react.jsx]: If this example file isn't really used, it should be removed.

File: app/views/layouts/mailer.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (alvarocavalcanti) [inline app/views/layouts/mailer.html.erb]: Is this file used?

File: app/views/user/User.html.erb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<h1>USER SIGNUP PAGE</h1>
```
Comment (alvarocavalcanti) [inline app/views/user/User.html.erb:1]: Should this file be used by the "create user page" use case?

File: config/initializers/application_controller_renderer.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
```
Comment (alvarocavalcanti) [inline config/initializers/application_controller_renderer.rb]: This file is entirely commented out. If it's not necessary should it be deleted?

File: config/initializers/content_security_policy.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
```
Comment (alvarocavalcanti) [inline config/initializers/content_security_policy.rb]: Another dead file

File: config/initializers/inflections.rb [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
```
Comment (alvarocavalcanti) [inline config/initializers/inflections.rb]: Another dead file

File: spec/views/user/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+
+
+RSpec.describe UserController, type: :controller do
+  describe 'GET #User' do
+    context 'when viewing homepage in the browser' do
```
Comment (alvarocavalcanti) [inline spec/views/user/user_spec.rb]: This context is confusing. It's parent reads `GET #User`, which implies that we're about to retrieve a single user, but the local context talks about viewing the homepage.

File: spec/views/user/user_spec.rb [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+
+
+RSpec.describe UserController, type: :controller do
```
Comment (alvarocavalcanti) [inline spec/views/user/user_spec.rb]: This is file is under the `spec/views` folder, and thus it should not test the `UserController`, this context should be moved to `spec/controllers/user_controller_spec.rb`.

File: test.rb [hunk 12]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (alvarocavalcanti) [inline test.rb]: According to the [documentation](https://docs.ruby-lang.org/en/3.0.0/doc/syntax/comments_rdoc.html#label-frozen_string_literal+Directive) this directive should be used only if there are any string literals being specified in the file, otherwise there's no need to have it here.

**Review by alvarocavalcanti — CHANGES_REQUESTED** [review]
Additionally to my comments below, the `.DS_Store` files should be added to `.gitignore` and removed from this PR.

**Comment by alvarocavalcanti** [comment]
Also, regarding the `.gitignore` file, have a look at the following examples:

* https://github.com/github/gitignore/blob/master/Rails.gitignore
* https://stackoverflow.com/a/62342961/1456509

File: app/assets/config/manifest.js (hunk 2 above)
Comment (jamesfairbairn-wpengine) [reply app/assets/config/manifest.js:1]: This is not dead code, from memory the `//=` is syntax I believe. So don't delete this.

Comment (EoinMcD) [reply app/assets/config/manifest.js:1]: Ive tried looking up if "//=" means anything, do you have any idea what it does?

File: app/assets/stylesheets/application.css (hunk 3 above)
Comment (jamesfairbairn-wpengine) [reply app/assets/stylesheets/application.css:1]: Yeah, leaving be might be the best for now, as I think this file is used to ensure css files get imported properly and served by the rails application properly.

Comment (alvarocavalcanti) [reply app/assets/stylesheets/application.css:1]: Sounds good 👍

File: app/assets/config/manifest.js (hunk 2 above)
Comment (jamesfairbairn-wpengine) [reply app/assets/config/manifest.js:1]: I think it's related to [this](https://github.com/rails/sprockets/blob/master/UPGRADING.md#manifestjs)

File: .rubocop_todo.yml [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+# Configuration parameters: EmptyLineBetweenMethodDefs, EmptyLineBetweenClassDefs, EmptyLineBetweenModuleDefs, AllowAdjacentOneLineDefs, NumberOfEmptyLines.
+Layout/EmptyLineBetweenDefs:
+  Exclude:
+    - 'test.rb'
+
+# Offense count: 2
+# Cop supports --auto-correct.
+Layout/EmptyLines:
+  Exclude:
+    - 'spec/views/user/user_spec.rb'
```
Comment (jamesfairbairn-wpengine) [inline .rubocop_todo.yml]: I think a lot of the cops being ignored in this file could be easily corrected by running `rubocop -a` to auto correct the issues. I'd be curious what happens if you delete this file and then run that command and then run `rubocop --auto-gen-config`

File: README.md (hunk 1 above)
Comment (EoinMcD) [reply README.md:1]: Updated the readme file, I wasn't sure what else to add

File: app/views/user/User.html.erb (hunk 6 above)
Comment (EoinMcD) [reply app/views/user/User.html.erb:1]: As discussed, no changes were made for this

File: spec/views/user/user_spec.rb (hunk 10 above)
Comment (EoinMcD) [reply spec/views/user/user_spec.rb]: Updated that, thanks

File: spec/views/user/user_spec.rb (hunk 11 above)
Comment (EoinMcD) [reply spec/views/user/user_spec.rb]: Moved to controllers folder, I originally had it testing the view but realised I should do it this way, which is why it was in that folder.

File: test.rb (hunk 12 above)
Comment (EoinMcD) [reply test.rb]: This file shouldn't have been included in the PR, it was me testing tests.

File: .rubocop_todo.yml (hunk 13 above)
Comment (EoinMcD) [reply .rubocop_todo.yml]: I did this, it recreated the file with less offense counts

File: README.md (hunk 1 above)
Comment (jamesfairbairn-wpengine) [reply README.md:1]: A few good things to add to a `ReadMe` in general are:
* How do I get it up and running?
* How do I run the test suite?
* Any special requirements or notes for development?

**Review by sbodwyer — COMMENTED** [review]
As a general comment try to reduce the scope of the PR. This PRs title suggest that you are adding a new page however you are checking in cop stylings, gitginore files etc. So as a suggestion you could have created a separate PR entitled 'setting up repo'  for all of that work. Perhaps no need to do now but be aware that the PR title should match the scope of what you are checking in

**Review by neildaly-wpengine — APPROVED** [review]
I'd agree with Shane's point above, but other than that, nice work overall in getting it set up! LGTM ⚡

**Review by alvarocavalcanti — APPROVED** [review]
LGTM
